### ISSUE TITLE
Display an error if a cart rule applies to all orders

### ISSUE THREAD

**Issue by lmeyer1** [PrestaShop/PrestaShop#28154]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

If someone inadvertently creates a cart rule without code and without any condition, it will apply to any order. It should not be possible to create such a cart rule. 

### Describe the solution you'd like

A cart rule should have at least one of these options configured:
- code
- customer
- country
- carrier
- customer group
- product rule
- shop association

If none of these is configured, an error should be raised. 

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by hibatallahAouadni** [comment]
Hello @lmeyer1 

> If someone inadvertently creates a cart rule without code and without any condition, it will apply to any order. It should not be possible to create such a cart rule.

I think some merchant may want to have a discount on all orders created at specific time (ex: on weekends) and that's logic!
For ex: there is a French cosmetics brand that offers free shipping and a gift for each order over 15 euros, and I can create as many orders as I want.

So, I think what you suggested may limit the choice of the merchant :confused: 
But, let's check with the product team if that's possible and acceptable :wink: 

Ping @PrestaShop/product-team wdyt about requesting at least one condition while creating a cart rule? 

Thanks in advance!

**Comment by lmeyer1** [comment]
@hibatallahAouadni I will think about it.
Your example states two conditions that I did not include in my implementation.
- _for each order over 15 euros_ => these are not all orders
- _at specific time (ex: on weekends)_ => if the user changes the default time that is filled automatically, then this is a condition.  Considering a start time in the future as a condition would be a good choice.

I will wait for more opinions before changing my implementation.

**Comment by MatShir** [comment]
Interesting, we need to be careful not to forget any use cases!
I imagine that some of the merchants you are working with have encountered the problem to report the bugs. It can be terrible a 15 euro discount for all order without condition and time limit, even for one night it can have a really bad effects on the revenues.
I looks okay but we need to discuss with @camilleganzin to be sure.

**Comment by camilleganzin** [comment]
Hello @lmeyer1, 
You may have a point here! But as @hibatallahAouadni said, there are some cases where merchants would want to apply discounts on all orders using Cart Rules. Rather than impose some limitations, we would prefer to alert the user when saving their cart rule that the discount will be applied to any orders since no conditions have been set.  
We can give you a template if you want to implement this solution.

**Comment by lmeyer1** [comment]
@camilleganzin You are right, an alert might be sufficient, but there is one gotcha:
If you create a cart rule, the default start time is _the last full hour_, i.e. if I create a cart rule at 17h55, it starts to be valid on 17h00.
No look at this scenario:
- I create by error a cart rule without any condition, it is immediately valid.
- I notice the alert you suggest
- A customer is just doing his checkout and the voucher applies
- I set up the cart rule and add the missing condition
- The customer got the reduction, and I was not even aware of it !

This is why I recommend rather:
- If the validity starts in the future, display a simple alert
- If the voucher would be immediately valid, display an error suggesting either adding a condition or to choose a validity in the future.

This solution solves both problems:
- It is easily possible to create a voucher without condition
- It is not possible to offer reductions by error to the next random order placed.

**Comment by camilleganzin** [comment]
@lmeyer1 you are right, I didn't think about that scenario. Let's do what you suggest!

**Comment by 04161999** [comment]
> ### Prerequisites
> 
> - [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
> - [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.
> 
> ### Is your feature request related to a problem?
> 
> If someone inadvertently creates a cart rule without code and without any condition, it will apply to any order. It should not be possible to create such a cart rule. 
> 
> ### Describe the solution you'd like
> 
> A cart rule should have at least one of these options configured:
> - code
> - customer
> - country
> - carrier
> - customer group
> - product rule
> - shop association
> 
> If none of these is configured, an error should be raised. 
> 
> ### Alternatives you've considered
> 
> _No response_
> 
> ### Additional context
> 
> _No response_
> 
> ### Do you plan to work on this feature?
> 
> - [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
> - [ ] I'm willing to provide any wireframes or design assets required for this feature.
> - [X] I'm willing to submit a Pull Request that implements this feature.
> - [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by lmeyer1** [comment]
@camilleganzin Thank you very much.

**Comment by kpodemski** [comment]
I'm changing the label as it was previously approved by PM

**Comment by Hlavtox** [comment]
I think this is a merchants problem, and limiting it will make most of the merchants angry. We cannot baby step people.

If I want to do 15% BLACK FRIDAY ON ALL ORDERS FOR EVERYONE, why should the software block me?
